### PR TITLE
feat(channels): Create posts in Twitter

### PR DESCRIPTION
**Pull request by ERosendo**
This PR adds support to use the Twitter API and fixes https://github.com/freelawproject/bigcases2/issues/26.

This PR introduces the following changes:

- Adds Twitter variables to the .env.example file and the setting module.

- Adds [twitterAPI ](https://github.com/geduldig/TwitterAPI) as a dependency.

- Adds a script to generate Twitter's token

- Adds an interface class (using protocols) called `BaseAPIConnector` to define the behavior of API connectors.

- Refactors the masto.py file. The current implementation of this utils file only uses functions, the new file implements a class called `MastodonConnector` and implements the methods specified in the `BaseAPIConnector` class.

- Adds a connector class to use the Twitter API

- Adds a new method to the channel model 
    - The new method is called `get_api_wrapper`. This new method returns an instance of the base API connector and uses the service field to handle different connectors.

- Tweaks the many-to-many relationship in the channel's model

- Adds a new selector to query all the enabled channels

- Renames the MastodonTemplate dataclass to use it as a base to create service-specific templates and also adds a class for Twitter and Mastodon posts.

- Moves all the template instances to a file called templates.py

- Update the imports in the test.py file

- Add border to the TextImage class

- Update type hints in the BaseAPIConnector class

- Updates method that handle webhook events and adds logic to create status updates in different channels ( Twitter and Mastodon) every time the bot gets a webhook event.

### CODE REVIEW

**Review by mlissner — COMMENTED** [review]
Looks pretty great and I learned a few things. A few tweaks, but let's make Twitter happen!

File: bc/channel/utils/connectors/twitter.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            settings.TWITTER_ACCESS_TOKEN_SECRET,
+            api_version=version,
+        )
+        return api
+
+    def add_status(self, message: str, text_image: TextImage | None) -> int:
+        """
+        Creates a new status update using the Twitter API.
+
+        The current implementation of the Twitter API(version 2) doesn't have an endpoint
```
[the remark this answers is not in the document]
Comment (ERosendo) [reply bc/channel/utils/connectors/twitter.py:31]: It's true. There's a table [here](https://developer.twitter.com/en/docs/twitter-api/migrate/twitter-api-endpoint-map) that maps the Twitter API v2 endpoints to the corresponding [standard v1.1](https://developer.twitter.com/en/docs/twitter-api/v1), [premium v1.1](https://developer.twitter.com/en/docs/twitter-api/premium), and [enterprise](https://developer.twitter.com/en/docs/twitter-api/enterprise) endpoints. 

This table is supposed to help with the migration to the new v2 API but They still have several items marked as **[COMING SOON]** and the media endpoints are one of them.

[the remark this answers is not in the document]
Comment (mlissner) [reply bc/channel/utils/connectors/twitter.py:31]: Insane. Maybe link to the table in our code, then for future folks to check again?

[the remark this answers is not in the document]
Comment (ERosendo) [reply bc/channel/utils/connectors/twitter.py:31]: That's a great idea.

File: bc/channel/utils/connectors/twitter.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                    "media": {"media_ids": [str(media_id)]},
+                },
+                method_override="POST",
+            )
+            data = response.json()
+        else:
+            response = self.api_v2.request(
+                "tweets", params={"text": message}, method_override="POST"
+            )
+            response.response.raise_for_status()
```
[the remark this answers is not in the document]
Comment (ERosendo) [reply bc/channel/utils/connectors/twitter.py:54]: You're right! I noticed it too. 

I also realized We were not adding **alt text** to the media file but I tried to use the library to attach it and It failed. 

I opened an [issue](https://github.com/geduldig/TwitterAPI/issues/224) and created a [PR ](https://github.com/geduldig/TwitterAPI/pull/225) in the [TwitterAPI ](https://github.com/geduldig/TwitterAPI) repo to fix the issue with the endpoint to add the **alt text**.

File: bc/core/utils/status/templates.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+)
+
+TWITTER_POST_TEMPLATE = TwitterTemplate(
+    link_placeholders=["pdf_link", "docket_link"],
+    str_template="""New filing: "{docket}"
+Doc #{doc_num}: {description}
+
+PDF: {pdf_link}
+Docket: {docket_link}""",
+)
```
[the remark this answers is not in the document]
Comment (ERosendo) [reply bc/core/utils/status/templates.py:41]: understood!

File: bc/channel/management/commands/post.py [hunk 4]
```diff
@@ -60,8 +60,8 @@ def handle(self, *args, **options):
                     )
                 )
 
-                m = get_mastodon()
-                m.status_post(post_text)
+                m = MastodonConnector()
+                m.add_status(post_text)
             else:
                 raise NotImplementedError(
```
[the remark this answers is not in the document]
Comment (ERosendo) [reply bc/channel/management/commands/post.py]: Yes, it is!

**Comment by mlissner** [comment]
Looks good. I just merged and I'll work on getting it deployed soon.

**Pull request merged**